### ISSUE TITLE
Synapse结果复现

### ISSUE THREAD

**Issue by llwwhh4002** [Cccccczh404/H-SAM#23]
我在复现过程中使用synapse的全部训练集进行训练，使用github中提供的训练命令进行训练后在测试集上的平均dice得分相差1.6个点（论文结果是86.49，复现结果是84.89），可以问一下可能是什么原因导致的吗，还是说训练过程中会有偶然性，需要多重复训练几次嘛

**Comment by hjklearn** [comment]
你好·，你是怎么复现的，图片输入大小是多少。

**Comment by mrbird643** [comment]
请问您有更改代码中的参数吗

**Comment by hjklearn** [comment]
> 请问您有更改代码中的参数吗
我是用项目参数训练的

**Comment by Mrhuhappy** [comment]
> 我在复现过程中使用synapse的全部训练集进行训练，使用github中提供的训练命令进行训练后在测试集上的平均dice得分相差1.6个点（论文结果是86.49，复现结果是84.89），可以问一下可能是什么原因导致的吗，还是说训练过程中会有偶然性，需要多重复训练几次嘛

近期我在复现 H-SAM 时遇到一些问题：在全监督设置下，我始终无法达到论文中报告的结果。 当前我复现得到的 DSC 最高约为 74%，但在部分设置下 DSC 甚至低于 10%，整体表现与论文结果差距较大。 我使用的数据集是 Synapse（从 TransUNet 的公开版本下载），并采用了代码中默认的训练配置：img_size=224、vit_name=vit_l、ckpt=sam_vit_l_0b3195.pth、rank=5。 在阅读论文后，我也尝试将 rank 调整为 4（目前仍在测试）。 数据方面，我也使用过 train_npz_new_224，但训练完成后效果仍不理想。 想请教您：在复现过程中，您是如何设置参数的，使 dsc 达到了 84.89？ 我的参数信息如下 ：parser = argparse.ArgumentParser()
parser.add_argument('--root_path', type=str,
default='/home/data/Synapse/train_npz', help='root dir for data')
parser.add_argument('--output', type=str, default='./output')
parser.add_argument('--dataset', type=str,
default='Synapse', help='experiment_name')
parser.add_argument('--list_dir', type=str,
default='/home/data/Synapse/lists/lists_Synapse', help='list dir')
parser.add_argument('--split', type=str,
default='train', help='list dir')
parser.add_argument('--num_classes', type=int,
default=8, help='output channel of network')
parser.add_argument('--max_iterations', type=int,
default=30000, help='maximum epoch number to train')
parser.add_argument('--max_epochs', type=int,
default=300, help='maximum epoch number to train')
parser.add_argument('--stop_epoch', type=int,
default=300, help='maximum epoch number to train')
parser.add_argument('--batch_size', type=int,
default=8, help='batch_size per gpu')
parser.add_argument('--n_gpu', type=int, default=1, help='total gpu')
parser.add_argument('--deterministic', type=int, default=1,
help='whether use deterministic training')
parser.add_argument('--base_lr', type=float, default=0.005,
help='segmentation network learning rate')
parser.add_argument('--img_size', type=int,
default=224, help='input patch size of network input')
parser.add_argument('--seed', type=int,
default=2345, help='random seed')
parser.add_argument('--vit_name', type=str,
default='vit_l', help='select one vit model')
parser.add_argument('--ckpt', type=str, default='checkpoints/sam_vit_l_0b3195.pth',
help='Pretrained checkpoint')
parser.add_argument('--lora_ckpt', type=str, default=None, help='Finetuned lora checkpoint')
parser.add_argument('--rank', type=int, default=4, help='Rank for LoRA adaptation')
parser.add_argument('--warmup', action='store_true', help='If activated, warp up the learning from a lower lr to the base_lr')
parser.add_argument('--warmup_period', type=int, default=250,
help='Warp up iterations, only valid when warmup is activated')
parser.add_argument('--AdamW', action='store_true', help='If activated, use AdamW to finetune SAM model')
parser.add_argument('--module', type=str, default='sam_lora_image_encoder')
parser.add_argument('--dice_param', type=float, default=0.9)
args = parser.parse_args()
